server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -240,7 +240,7 @@ public ShardSearchRequest(
         this.originalIndices = originalIndices;
         this.readerId = readerId;
         this.keepAlive = keepAlive;
-        assert keepAlive == null || readerId != null : "readerId: " + readerId + " keepAlive: " + keepAlive;
+        assert keepAlive == null || readerId != null : "readerId: null keepAlive: " + keepAlive;
```

Comment:
I don't understand this.

This assertion is tripped if:
 - keepAlive is `null`
 - OR `readerId` ISN'T `null`.

Why do we replace readerId in the string builder with `null`? Is it always passed in as null in the ctor?

Reply:
If keep alive is not null, then the assertion is only tripped if the reader is null.
If keep alive is null, the assertions isn't tripped.
=> reader is always null if assertion is tripped :)

Reply:
@original-brownbear 🤦 yes, I was reading it backwards!